### PR TITLE
[PDI-12176] - "Execute a shell script" job fails with "No such file or directory" (when developed on Windows, executed on Linux)

### PR DESCRIPTION
**Pull request by YuryBY**
@mattcasters, @brosander, would you mind of reviewing/merging

### CODE REVIEW

**Comment by YuryBY** [comment]
@mattyb149 could you review/merge please

File: engine/src/org/pentaho/di/job/entries/shell/JobEntryShell.java [hunk 1]
```diff
@@ -675,6 +682,16 @@ private FileObject createTemporaryShellFile( FileObject tempFile, String fileCon
     return tempFile;
   }
 
+  @VisibleForTesting
+  String replaceWinEOL( String input ) {
+    String result = input;
+    // replace Windows's EOL if it's contained ( see PDI-12176 )
+    result = result.replaceAll( "\\r\\n", "\n" );
+    result = result.replaceAll( "\\r", "\n" );
```
Comment (mattyb149) [inline engine/src/org/pentaho/di/job/entries/shell/JobEntryShell.java]: This should be able to be combined into a single regex ("\\r\\n?")

File: engine/test-src/org/pentaho/di/job/entries/shell/JobEntryShellTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package org.pentaho.di.job.entries.shell;
```
Comment (mattyb149) [inline engine/test-src/org/pentaho/di/job/entries/shell/JobEntryShellTest.java]: Needs a copyright/license header

**Comment by YuryBY** [comment]
@mattyb149, PR is updated. Thanks

**Comment by YuryBY** [comment]
@mattcasters, @brosander, @mattyb149, any updates?

**Pull request merged**